### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by tumainimaganiko**
In this project we did the following:

#### Logic

- Each class from the [UML class diagram](./images/catalog_of_my_things.png) is created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are implemented.
- All properties visible in the diagram are defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships are NOT set in the constructor method. Instead, they have a custom setter method created._
- All methods visible in the diagram are implemented:
    - add_item method (in all classes that have that method)
        - Takes an instance of the Item class as an input.
        - Adds the input item to the collection of items.
        - Adds self as a property of the item object (by using the correct setter from the item object).
    - can_be_archived?() in the Item class
        - Returns true if published_date is older than 10 years.
        - otherwise, it returns false.
    - move_to_archive() in the Item class
        - Reuses can_be_archived?() method.
        - It changes the archived property to true if the result of the can_be_archived?() method is true.
        - It does nothing if the result of the can_be_archived?() method is false.
    - can_be_archived?() in the Book class
        - Overrides the method from the parent class.
        - Returns true if parent's method returns true OR if cover_state equals to "bad".
        - otherwise, it returns false.
    - can_be_archived?() in the MusicAlbum class
        - It overrides the method from the parent class.
        - Returns true if parent's method returns true AND if on_spotify equals true.
        - otherwise, it returns false.
    - can_be_archived?() in the Movie class
        - It overrides the method from the parent class.
        - Returns true if parent's method returns true OR if silent equals true.
        - otherwise, it returns false.
    - can_be_archived?() in the Game class
        - It overrides the method from the parent class.
        - It returns true if parent's method returns true AND if last_played_at is older than 2 years.
        - otherwise, it returns false.
- Added unit tests for all implemented methods.

#### User interface

- We created a main.rb file that will serve as our console app entry point.
 - All data is preserved by saving collections in .json files.

### Database
- We created a schema.sql file with tables that will be analogical to the structure of the classes in our app.

We have added the video of presentation in the Readme file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,144 @@
...
+<details>
+  <summary>Ruby</summary>
+  
+</details>
+
+<!-- prsentation vidéo -->
+### Video presentation 
+[Video](https://drive.google.com/file/d/1w_kzPfXvjtkez7Z3yPGWOS_uLTW-9eGy/view?usp=sharing)
+
+<!-- Features -->
```
Comment (thecodechaser) [inline README.md:41]: - [ ] Please make sure the max length of the presentation video is 5 minutes as requested in the project requirements. Currently, your presentation video is 6+ minutes long. Please fix this issue.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @rachaouldbabaali, and Team👋, 

### You are doing a marvelous job on the Ruby group capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests passed. ✔️
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
Thank you @thecodechaser, we truly appreciate your time and efforts in reviewing our work

**Review by MahmoudBakr23 — APPROVED** [review]
Hi, team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done modifying the presentation video as requested by the previous reviewer

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
Thank you @MahmoudBakr23 , we truly appreciate your time and efforts in reviewing our work

**Pull request merged**